demo/runtime/src/runtime/council.rs

```diff
@@ -0,0 +1,1236 @@
+	storage::exists(&who.to_keyed_vec(REGISTER_INFO_OF))
+}
+
+/// Determine the block that a vote can happen on which is no less than `n`.
+pub fn next_vote_from(n: BlockNumber) -> BlockNumber {
+	let voting_period = voting_period();
+	(n + voting_period - 1) / voting_period * voting_period
+}
+
+/// The block number on which the tally for the next election will happen. `None` only if the
+/// desired seats of the council is zero.
+pub fn next_tally() -> Option<BlockNumber> {
+	let desired_seats = desired_seats();
+	if desired_seats == 0 {
+		None
+	} else {
+		let c = active_council();
+		let (next_possible, count, coming) =
+			if let Some((tally_end, comers, leavers)) = next_finalise() {
+				// if there's a tally in progress, then next tally can begin immediately afterwards
+				(tally_end, c.len() - leavers.len() + comers as usize, comers)
+			} else {
+				(system::block_number(), c.len(), 0)
+			};
+		if count < desired_seats as usize {
+			Some(next_possible)
+		} else {
+			// next tally begins once enough council members expire to bring members below desired.
+			if desired_seats <= coming {
+				// the entire amount of desired seats is less than those new members - we'll have
+				// to wait until they expire.
+				Some(next_possible + term_duration())
+			} else {
+				Some(c[c.len() - (desired_seats - coming) as usize].1)
+			}
+		}.map(next_vote_from)
+	}
+}
+
+/// The accounts holding the seats that will become free on the next tally.
+pub fn next_finalise() -> Option<(BlockNumber, u32, Vec<AccountId>)> {
+	storage::get(NEXT_FINALISE)
+}
+
+/// The total number of votes that have happened or are in progress.
+pub fn vote_index() -> VoteIndex {
+	storage::get_or_default(VOTE_COUNT)
+}
+
+/// The present candidate list.
+pub fn candidates() -> Vec<AccountId> {
+	storage::get_or_default(CANDIDATES)
+}
+
+/// The present voter list.
+pub fn voters() -> Vec<AccountId> {
+	storage::get_or_default(VOTERS)
+}
+
+/// The vote index and list slot that the candidate `who` was registered or `None` if they are not
+/// currently registered.
+pub fn candidate_reg_info(who: &AccountId) -> Option<(VoteIndex, u32)> {
+	storage::get(&who.to_keyed_vec(REGISTER_INFO_OF))
+}
+
+/// The last cleared vote index that this voter was last active at.
+pub fn voter_last_active(voter: &AccountId) -> Option<VoteIndex> {
+	storage::get(&voter.to_keyed_vec(LAST_ACTIVE_OF))
+}
+
+/// The last cleared vote index that this voter was last active at.
+pub fn approvals_of(voter: &AccountId) -> Vec<bool> {
+	storage::get_or_default(&voter.to_keyed_vec(APPROVALS_OF))
+}
+
+/// Get the leaderboard if we;re in the presentation phase.
+pub fn leaderboard() -> Option<Vec<(Balance, AccountId)>> {
+	storage::get(LEADERBOARD)
+}
+
+pub mod public {
+	use super::*;
+
+	/// Set candidate approvals. Approval slots stay valid as long as candidates in those slots
+	/// are registered.
+	pub fn set_approvals(signed: &AccountId, votes: &Vec<bool>, index: VoteIndex) {
```

Comment:
should be a slice

Reply:
would require either substantial changes to `storage` (in order to use `AsRef` trait to realise that the slice should be encoded as a `Vec`) or a copy.

Reply:
ah, yeah...the storage API needs an overhaul for sure then.